Program.cs

```diff
@@ -1,143 +1,83 @@
-        }
-        catch (Exception e)
-        {
-            Console.Error.WriteLine($"Error: {e.Message}");
-            return null;
-        }
-
-        if (response.IsSuccessStatusCode)
-        {
-            string responseBody = await response.Content.ReadAsStringAsync();
-            var json = JObject.Parse(responseBody);
-            return json["upload_url"].ToString();
-        }
-        else
-        {
-            Console.Error.WriteLine($"Error: {response.StatusCode} - {response.ReasonPhrase}");
-            return null;
-        }
-    }
-}
-
-// Upload a file and create a transcript using the AssemblyAI API
-public class TranscriptFetcher
-{
-    // Function to fetch transcript asynchronously
-    public async Task<dynamic> GetTranscriptAsync(string apiToken, string audioUrl)
-    {
-        string url = "https://api.assemblyai.com/v2/transcript";
-
-        var data = new Dictionary<string, string>()
-        {
-            { "audio_url", audioUrl }
-        };
-
-        using (var client = new HttpClient())
-        {
-            client.DefaultRequestHeaders.Add("authorization", apiToken);
-
-            var content = new StringContent(JsonConvert.SerializeObject(data), Encoding.UTF8, "application/json");
-
-            HttpResponseMessage response = await client.PostAsync(url, content);
-
-            var responseContent = await response.Content.ReadAsStringAsync();
 
-            var responseJson = JsonConvert.DeserializeObject<dynamic>(responseContent);
-
-            string transcriptId = responseJson.id;
-
-            string pollingEndpoint = $"https://api.assemblyai.com/v2/transcript/{transcriptId}";
-
-            while (true)
-            {
-                var pollingResponse = await client.GetAsync(pollingEndpoint);
-
-                var pollingResponseContent = await pollingResponse.Content.ReadAsStringAsync();
-
-                var pollingResponseJson = JsonConvert.DeserializeObject<dynamic>(pollingResponseContent);
-
-                if (pollingResponseJson.status == "completed")
-                {
-                    return pollingResponseJson;
-                }
-                else if (pollingResponseJson.status == "error")
-                {
-                    throw new Exception($"Transcription failed: {pollingResponseJson.error}");
-                }
-                else
-                {
-                    Thread.Sleep(3000);
-                }
-            }
-        }
-    }
-}
+namespace AssemblyAiConsole;
 
 public class Program
 {
+    private static readonly HttpClient HttpClient = new HttpClient();
+
     public static async Task Main(string[] args)
     {
         // Your API token is already set in this variable
-        string apiToken = Environment.GetEnvironmentVariable("ASSEMBLY_AI_API_KEY"); // only pre-modification for git
+        var apiToken = Environment.GetEnvironmentVariable("ASSEMBLY_AI_API_KEY"); // only pre-modification for git
```

Comment:
This was hardcoded before which may be dangerous. Devs may accidentally upload their code to a public source control repository, or embed their API key inside a client distributed application.

.NET dev's don't really use environment variables for local development, but setting up a configuration builder and using user-secrets may be too much work for this copy/paste sample.